src/sizzle.js

```diff
@@ -671,8 +671,10 @@ setDocument = Sizzle.setDocument = function( node ) {
 
 					// Fall back on getElementsByName
 					elems = context.getElementsByName( id );
+					length = elems.length;
```

Comment:
NodeList#length can be an element in IE<9 (and we should have a test where that is the case on this line).

Reply:
So, something like this?

```js
if (typeof elem.length === 'number') {
  fastPath();
} else {
  slowPathForLegacyBrowsers();
}
```

Reply:
Yes, provided we can get it without excessive file size increase.

Reply:
[example](https://github.com/jquery/sizzle/blob/eabce51ea7360c4507da2eeaec2633378de4ec8d/src/sizzle.js#L1995):
```js
for ( ; i !== len && (elem = elems[i]) != null; i++ ) {
```